service/policy/actions/actions_test.go

```diff
@@ -0,0 +1,273 @@
+package actions
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/bufbuild/protovalidate-go"
+	"github.com/opentdf/platform/protocol/go/common"
+	"github.com/opentdf/platform/protocol/go/policy"
+	"github.com/opentdf/platform/protocol/go/policy/actions"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	validUUID                    = "00000000-0000-0000-0000-000000000000"
+	errMessageUUID               = "string.uuid"
+	errMessageMaxLength          = "string.max_len"
+	errMessageRequiredActionName = "action_name_format"
+	errMessageOptionalActionName = "action_name_format"
+	errMessageURI                = "string.uri"
+	errMessageRequired           = "required"
+)
+
+var (
+	validNames = []string{"valid_Name", "valid_name", "NAME", "NAME-IS-VALID", "SOME_VALID_NAME", "valid-name", strings.Repeat("a", 253), strings.Repeat("a", 1)}
+
+	invalidNameTests = []string{
+		"!",
+		"name with space",
+		"slash/",
+		"slash\\",
+		"name:with:colon",
+		"name.dot.delimited",
+		"_cannot_start_with_underscore",
+		"cannot_end_with__underscore_",
+		"-cannot-start-with-hyphen",
+		"cannot-end-with-hyphen-",
+	}
+)
+
+// Actions proto validation
+
+type ActionSuite struct {
+	suite.Suite
+	v protovalidate.Validator
+}
+
+// Set up the test environment
+func (s *ActionSuite) SetupSuite() {
+	v, err := protovalidate.New()
+	if err != nil {
+		panic(err)
+	}
+	s.v = v
+}
+
+func TestActionsServiceProtos(t *testing.T) {
+	suite.Run(t, new(ActionSuite))
+}
+
+func (s *ActionSuite) Test_CreateActionRequest_Fails() {
+	for _, name := range invalidNameTests {
+		n := name // Capture the loop variable locally
+		s.Run(n, func() {
+			req := &actions.CreateActionRequest{
+				Name: n,
+			}
+			err := s.v.Validate(req)
+			s.Require().Error(err)
+			s.Require().Contains(err.Error(), errMessageRequiredActionName)
+		})
+	}
+
+	// no name
+	req := &actions.CreateActionRequest{
```

Comment:
Nit: another way of doing this would be adding the expected err message and test name to a testcases struct: https://github.com/opentdf/platform/pull/2066/files#diff-7308d2e883c56c9eed1f3372e0289a3238fe9b5996998b38d26453a34814f7abR80-R167

Reply:
Thanks, and agreed. I opted to iterate through the slice of names that fail due to the invalid name format validation and keep other tests separate as they differ if the name in the request proto is optional or not (identifier name in GET and updated name in UPDATE are optional).

Reply:
However, I did push up some a new commit to clean up some unit test variable names.